src/client/editor/project.js

```diff
@@ -241,16 +243,33 @@ let projectTab = function() {
 			while(Object.keys(prefabsTab.prefabs).length > 0) {
 				prefabsTab.deletePrefab(Object.keys(prefabsTab.prefabs)[0]);
 			}
-			while(Object.keys(modelsTab.models).length > 0) {
-				modelsTab.removeModel(Object.keys(modelsTab.models)[0]);
+			let modelsKeys = Object.keys(modelsTab.models);
+			for (let i = modelsKeys.length - 1; i >= 0; i--) {
+				modelsTab.models[modelsKeys[i]].delete();
+			}
+			let materialsKeys = Object.keys(materialsTab.materials);
+			for (let i = materialsKeys.length - 1; i >= 0; i--) {
+				materialsTab.materials[materialsKeys[i]].delete();
+			}
+			while (Object.keys(texturesTab.textures).length > 0) {
+				texturesTab.removeTexture(Object.keys(texturesTab.textures)[0]);
 			}
 
 			//Initialize loaded project
 			editorLog("Loading project...");
 			this.activeProject = loadedProject;
 
+			for (let key in this.activeProject.textures) {
+				texturesTab.addTexture(key, this.activeProject.textures[key].name, this.activeProject.textures[key].file, this.activeProject.textures[key]);
```

Comment:
I'm *decently* sure that it will remain synchronous, as the TextureLoader generates a `Texture` and returns it synchronously. It just updates it once it has actually loaded. But thanks for the reminder!

Reply:
I get the feeling the loading of the image is done synchronously as well, since it seems to block the main thread while loading a project (which isn't the case with models I don't think, which also loads textures..?). It's not a huge issue, but we might want to look at it eventually, maybe. Very maybe.